Patch-Source: https://sourceforge.net/p/mcj/fig2dev/ci/8425b790b4656ea240430062ec51bd2334ceff11/
Included with the application of detect-nan-in-spline-control-values.patch.
---
--- a/fig2dev/read.c
+++ b/fig2dev/read.c
@@ -1576,9 +1576,11 @@ read_splineobject(FILE *fp, char **restr
 		if (lx < INT_MIN || lx > INT_MAX || ly < INT_MIN ||
 				ly > INT_MAX || rx < INT_MIN || rx > INT_MAX ||
 				ry < INT_MIN || ry > INT_MAX) {
-			/* do not care to clean up, we exit anyway
-			   cp->next = NULL;
-			   free_splinestorage(s);	*/
+			/* clean up, to pass test "reject huge spline controls
+			   values" when -fsanitize=address is enabled */
+			cp->next = NULL;
+			free_splinestorage(s);
+			free(cq);
 			put_msg("Spline control points out of range "
 					"at line %d.", *line_no);
 			exit(EXIT_FAILURE);
